File: testdata/dataflow_sample.py
import json
import pickle

def handle_request(request):
    body = request.data
    payload = json.loads(body)
    name = payload["name"]
    query = "SELECT * FROM users WHERE name = " + name
    return query

def simple_assignment():
    x = "hello"
    y = x
    return y

def field_access(user):
    email = user.email
    return email

class UserService:
    def create_user(self, data):
        name = data["name"]
        result = self.db.execute(name)
        return result

File: testdata/dataflow_sample.ts
import express from 'express';

function handleRequest(req: express.Request, res: express.Response): void {
    const body = req.body;
    const payload = JSON.parse(body);
    const name: string = payload.name;
    const query = "SELECT * FROM users WHERE name = " + name;
    res.send(query);
}

function simpleAssignment(): string {
    const x = "hello";
    const y = x;
    return y;
}

function fieldAccess(user: { email: string }): string {
    const email = user.email;
    return email;
}

const arrowFn = (data: string): string => {
    const result = data.trim();
    return result;
};
